repose-aggregator/components/services/container-configuration-service/impl/src/main/scala/org/openrepose/nodeservice/containerconfiguration/DeploymentConfigPatchUtil.scala

```diff
@@ -0,0 +1,155 @@
+    * @return a new configuration object with the patch applied to the base
+    */
+  def patchLoggingConfiguration(base: LoggingConfiguration,
+                                patch: LoggingConfiguration): LoggingConfiguration = {
+    Option(patch.getHref).foreach(base.setHref)
+    Option(patch.getValue).foreach(base.setValue)
+
+    base
+  }
+
+  /**
+    * This function does not return a copy of the base configuration, but instead mutates it directly.
+    *
+    * @param base  the base configuration object
+    * @param patch the patch configuration object, the values of which will override the base
+    * @return a new configuration object with the patch applied to the base
+    */
+  def patchDeploymentDirectory(base: DeploymentDirectory,
+                               patch: DeploymentDirectoryPatch): DeploymentDirectory = {
+    Option(patch.getValue).foreach(base.setValue)
+    Option(patch.isAutoClean).foreach(base.setAutoClean)
+
+    base
+  }
+
+  /**
+    * This function does not return a copy of the base configuration, but instead mutates it directly.
+    *
+    * @param base  the base configuration object
+    * @param patch the patch configuration object, the values of which will override the base
+    * @return a new configuration object with the patch applied to the base
+    */
+  def patchArtifactDirectory(base: ArtifactDirectory,
+                             patch: ArtifactDirectoryPatch): ArtifactDirectory = {
+    Option(patch.getValue).foreach(base.setValue)
+    Option(patch.getCheckInterval).foreach(base.setCheckInterval)
+
+    base
+  }
+  */
+
+  /**
+    * This function does not return a copy of the base configuration, but instead mutates it directly.
+    *
+    * @param base  the base configuration object
+    * @param patch the patch configuration object, the values of which will override the base
+    * @return a new configuration object with the patch applied to the base
+    */
+  def patchSslConfiguration(base: SslConfiguration,
+                            patch: SslConfigurationPatch): SslConfiguration = {
+    Option(patch.getKeystoreFilename).foreach(base.setKeystoreFilename)
+    Option(patch.getKeystorePassword).foreach(base.setKeystorePassword)
+    Option(patch.getKeyPassword).foreach(base.setKeyPassword)
+    Option(patch.getTruststoreFilename).foreach(base.setTruststoreFilename)
+    Option(patch.getTruststorePassword).foreach(base.setTruststorePassword)
+    Option(patch.getIncludedProtocols).foreach(base.setIncludedProtocols)
+    Option(patch.getExcludedProtocols).foreach(base.setExcludedProtocols)
+    Option(patch.getIncludedCiphers).foreach(base.setIncludedCiphers)
+    Option(patch.getExcludedCiphers).foreach(base.setExcludedCiphers)
+    Option(patch.isTlsRenegotiationAllowed).foreach(base.setTlsRenegotiationAllowed)
+    Option(patch.isNeedClientAuth).foreach(base.setNeedClientAuth)
+
+    base
+  }
+
+  private def deepCopy[T <: java.io.Serializable](obj: T): T = {
```

Comment:
So my other idea was just performing a shallow copy at every level when we apply the patch. With the way these methods are setup, it's relatively easy, but requires JAXB to make its generated classes `Cloneable`. Well that, or we always construct a new object, then populate the field either with the patch value, or the base value if the patch value does not exist.